1.16.x Updates

ChampionAsh5357 opened the pull request:
This contains an accurate update to forge docs as of 32.0.70 with mapping snapshot 20200514-1.16. Once again, I know I am merging this to the 1.15.x docs. There isn't one for 1.16.x.

This PR is made to address the issues and recommendations of tterrag1098 in #313. In line with that idea, I have split this PR into the three versions it changes: 1.14.x/1.15.x (#325) and this current one.

This PR only contains 1.16.x specific changes. That's why it may seems as alot of the files still reflect their original counterparts. All the version specific changes are each within their own PR which would combine together to make the original completed version.

Most of the changes in the original PR have been changed in 1.14.x. 1.15.x contained rendering changes and some minor fixes here and there that occurred between 1.14.x and 1.16.x. This PR focuses on the changing of mapping names, the use of AbstractBlock, the replacement of IProperty with Property, and The property override system becoming its own separate client.

To also do a complete match, the example within oak_log.json has been replaced with its data generated form. Since there is no primer written for 1.16 yet, the porting file has been removed until then.

Let me know if there is anything that needs updating or fixing.

ChampionAsh5357 committed:
Changes specific to 1.15.x

ChampionAsh5357 committed:
Reset to current 1.15.x

ChampionAsh5357 committed:
Fixes to match

ChampionAsh5357 committed:
All 1.16.x changes

tterrag1098 commented on docs/legacy/index.md:
```diff
@@ -16,3 +16,4 @@ Unfortunately not all versions were used for a significant amount of time and th
 |:-------------:|:----------:|:------------------------------------------|
 |    1.12.x     |   100%     | https://mcforge.readthedocs.io/en/1.12.x/ |
 |    1.13.x     |    10%     | https://mcforge.readthedocs.io/en/1.13.x/ |
+|    1.15.x     |   100%     | https://mcforge.readthedocs.io/en/1.15.x/ |
```
1.15.x is not legacy yet.

tterrag1098 commented on docs/legacy/porting1214.md:
```diff
@@ -1,6 +0,0 @@
-Porting from Minecraft 1.12 to Minecraft 1.13/1.14
-==================================================
-
-Due to major rewrites to Forge happening during the lifetime of 1.13 it never saw much usage and instead most mods ported directly to 1.14 skipping over 1.13. Therefore this guide will focus on porting from 1.12 directly to 1.14.
-
-_Full documentation on porting from 1.12 to 1.14 is still work in progress, for now we suggest having a look at [a primer written by williewillus](https://gist.github.com/williewillus/353c872bcf1a6ace9921189f6100d09a)!_
```
I don't see the point in removing this without replacement. These guides are still useful to someone coming from versions before 1.15.

tterrag1098 commented on docs/models/advanced/ibakedmodel.md:
```diff
@@ -22,7 +22,7 @@ If the model is being rendered as an item in an inventory, on the ground as an e
     Unless you know what you're doing and are OK with using deprecated features, just `return false` from this and continue on.
 
-When rendering this as an item, returning `true` causes the model to not be rendered, instead falling back to `TileEntityItemStackRenderer::renderItem`. For certain vanilla items such as chests and banners, this method is hardcoded to copy data from the item into a `TileEntity`, before using a `TileEntitySpecialRenderer` to render that TE in place of the item. For all other items, it will use the ...
+When rendering this as an item, returning `true` causes the model to not be rendered, instead falling back to `ItemStackTileEntityRenderer::func_239207_a_`. For certain vanilla items such as chests and banners, this method is hardcoded to copy data from the item into a `TileEntity`, before using a `TileEntitySpecialRenderer` to render that TE in place of the item. For all other items, it will use ...
```
I don't think this change is a good idea, it will be remapped pretty soon and then we'll have to change it back. Let's hold off on baking in new mappings until we have a RB.

tterrag1098 commented on docs/models/advanced/itemoverridelist.md:
```diff
@@ -11,9 +11,9 @@ Given a list of `ItemOverride`s, the constructor copies that list and stores the
 This is a deprecated vanilla method. It is only called in the vanilla `handleItemState`, and in almost all cases can be safely ignored.
 
-### `handleItemState`
+### `func_239290_a_`
```
Same here

tterrag1098 reviewed (changes requested):
I think it might be best to hold off on some of the name changes until 1.16 has stabilized a bit, perhaps this PR can remain open as an omnibus for these kinds of changes until RB is out.

ChampionAsh5357 committed:
PR Fixes from tterrag1098

Remove 1.15.x as not legacy

Added back port 1214.md

Removed mapping changes on ibakedmodel and itemoverridelist

ChampionAsh5357 commented:
Is this also the case for mappings in ItemModelProperties replacing addPropertyOverride? Or should I just leave them be and update when mapped since it does replace the entire system.

tterrag1098 commented:
As 1.16.x now has a RB out, this can be accepted once it's up to date.

ChampionAsh5357 committed:
Merge branch '1.16.x' into 1.16.x-new

ChampionAsh5357 committed:
Remap methods

ChampionAsh5357 commented:
@tterrag1098 Should be already to go whenever you can. Most of the information that's changed is still present in the other PR I have open, so it might be a good idea to review and merge that first before this one.
